src/mono/mono/mini/mini-amd64.c

```diff
@@ -2368,6 +2400,9 @@ mono_arch_get_llvm_call_info (MonoCompile *cfg, MonoMethodSignature *sig)
 		linfo->ret.storage = LLVMArgVtypeRetAddr;
 		linfo->vret_arg_index = cinfo->vret_arg_index;
 		break;
+	case ArgSwiftValuetypeLoweredRet:
+		// LLVM compilation of P/Invoke wrappers is not supported
```

Comment:
What does this mean? Can you add a `g_assert_not_reached` here if you don't expect to get here when llvm AOT compilation is used

Reply:
We can't assert here because llvm retrieves the call info before aborting due to pinvoke wrappers.
1. https://github.com/dotnet/runtime/blob/ab013a35ac6d76320bb6e28a49d62ad82a88e5d6/src/mono/mono/mini/mini-llvm.c#L13012
2. https://github.com/dotnet/runtime/blob/ab013a35ac6d76320bb6e28a49d62ad82a88e5d6/src/mono/mono/mini/mini-llvm.c#L13112-L13115

I think it should be possible to move (2) before (1) but I wasn't 100% confident that it wouldn't break some edge cases. Therefore I made this hack which avoids the assert but relies on the (2) that the pinvoke wrapper won't be llvm compiled.